components/MintControls/index.tsx

```diff
@@ -131,56 +152,73 @@ export default function MintControls({ pageState, setPageState }: MintControlsPr
     [network, contractAddress, valentineId]
   )
 
+  const earlyMintComplete = useMemo(() => hasEarlyMinted && !publicMintLive, [hasEarlyMinted, publicMintLive])
+
+  const buttonDisabled = useMemo(() => {
+    if (!accountData) {
+      return false
+    }
+    switch (pageState) {
+      case PAGE_STATE.PENDING:
+        return true
+      case PAGE_STATE.COMPLETE:
+      case PAGE_STATE.ERROR:
+      case PAGE_STATE.READY:
+        return earlyMintComplete || !readyToMint
+    }
+  }, [accountData, earlyMintComplete, pageState, readyToMint])
+
+  const buttonText = useMemo(() => {
+    if (!accountData) {
+      return 'CONNECT WALLET'
+    }
+    switch (pageState) {
+      case PAGE_STATE.COMPLETE:
+        return 'SEND ANOTHER'
+      case PAGE_STATE.ERROR:
+        return 'RETRY'
+      case PAGE_STATE.PENDING:
+        return 'PENDING...'
+      case PAGE_STATE.READY:
+        if (earlyMintComplete) {
+          return 'EARLY MINT CLAIMED!'
+        }
+        if (!publicMintLive) {
+          if (!earlyMintLive && !isEarlyMintLive) {
+            return 'MINT NOT OPEN YET'
+          }
+        }
+        return `MINT ${mintEthPrice.toString()} ETH`
```

Comment:
would be nice to get some info about *when* minting will be available here... maybe link to your tweet?